src/wallet/bdk.rs

```diff
@@ -308,77 +366,205 @@ impl Wallet {
                 let locktime = bond.lock_time;
                 println!(
                     "[{}] locktime={}",
-                    Address::from_script(&bond.script_pub_key(), self.store.network).unwrap(),
+                    Address::from_script(&bond.script_pub_key(), self.network).unwrap(),
                     locktime
                 );
             }
         }
         Ok(())
     }
 
-    pub fn init(
+    pub fn init<E: IntoWalletDescriptor + Hash + Debug>(
         path: &PathBuf,
-        rpc_config: &RPCConfig,
         seedphrase: String,
         passphrase: String,
-    ) -> Result<Self, WalletError> {
+        network: Network,
+        external_descriptor: E,
+        internal_descriptor: E,
+        fidelity_descriptor: E,
+        swapcoin_descriptor: E,
+        contract_descriptor: E,
```

Comment:
The user should be able to init the wallet with just seed and passphrase. All these descriptors should be derivable from the master seed and passphrase. Asking for descriptor and master seed together is very redundant, and will be a confusing API.